src/classes/CRLP_Debug_UTIL.cls

```diff
@@ -1,37 +1,100 @@
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
 /**
-* ====================================
-* TODO REMOVE THIS CLASS EVENTUALLY
-* ====================================
-* TEMPORARY STATIC VALUE AND METHODS TO HOLD CURRENT STATE TO USE FOR ROLLUP DEBUGGING
+* @author Salesforce.org
+* @date 2018
+* @group Customizable Rollups Debug Logger
+* @description TEMPORARY USE ONLY
 */
 public class CRLP_Debug_UTIL {
 
-    private static final Boolean isEnabled = false;
+    public static final Integer LEVEL_MIN = 1;
+    public static final Integer LEVEL_MAX = 2;
+
+    /*
+     * @description Rather than package the Customizable_Rollup_Settings__c.Debug_Level__c field, which will make it harder
+     * to remove at a later point, this requries that an admin (or sfdo staff) manually create the new custom settings field
+     * on this object as an unmanaged field. The code below attempts to check for the field and if it exists will use it
+     * to set the current debugging level. No field or a value of 0 or null means there CRLP Debugging is disabled.
+     * Level 0 = OFF
+     * Level 1 = MINIMAL
+     * Level 2 = DETAILED
+     */
+    public static Integer debuggingLevel {
+        get {
+            if (debuggingLevel == null) {
+                Customizable_Rollup_Settings__c crlpSettings = UTIL_CustomSettingsFacade.getCustomizableRollupSettings();
+                try {
+                    debuggingLevel = ((Double) crlpSettings.get('Debug_Level__c')).intValue();
```

Comment:
Should we package Debug_Level__c? Will we ever need it to be an integer, or can we just make it a checkbox? @force2b  (cc @bethbrains)

Reply:
I like it as a (default null) integer so someday we could actually have different levels based on the integer value. (1=light logging, 2=medium logging, 3=omgwhat'sgoingonlogallthethings)

Reply:
Actually it already supports 2 levels:
* 0/null = disabled
* 1=Level 1 debug (only write a summary at the end of the rollups)
* 2=Level 2 debug (collect more information to write at the end, but will have an impact on performance).

Reply:
I would suggest using Text instead of Integer for the Debug_Level__c field. That way, it will be more readable to developer/admin. So the reader won't need to look for what is the meaning of the numbers. Also, in the code, we can use Enums instead of Integer. I'm thinking about something like:

public enum DebuggingLevel {OFF, MINIMAL, DETAILED}

To be more explicit, should we remove the DebuggingLevel OFF, and have a field Active__c (checkbox) instead?

Reply:
I don't foresee this being something we'll ever want to give directly to customers to use. It's been built specifically for our own debugging use only. Thus the reason to not package the field. The class captures debug data, but what it generates will not be helpful at all to customer admins. I'd much prefer to keep this both extremely simple and somewhat obfuscated from customer admins.